frontend/src/components/DataTable.vue

```diff
@@ -24,7 +24,7 @@ interface Props {
   dataList: TableDataRow[], // List of data to be displayed
   filters?: TableFilter[], // List of filters to be displayed
   loading: boolean, // Displays a loading spinner
-  showPagination: boolean;
+  showPagination?: boolean;
```

Comment:
There was an issue with some components not having this attribute specified. Since it already had a default, I made it optional.